demo/admin/src/products/ProductsPage.tsx

```diff
@@ -42,7 +45,27 @@ const ProductsPage: React.FC = () => {
             <StackSwitch initialPage="grid">
                 <StackPage name="grid">
                     <StackToolbar scopeIndicator={<ContentScopeIndicator global />} />
-                    <ProductsGrid />
+                    <MainContent fullHeight>
+                        <ProductsGrid
+                            toolbarAction={
+                                <Button
+                                    startIcon={<AddIcon />}
+                                    component={StackLink}
+                                    pageName="add"
+                                    payload="add"
+                                    variant="contained"
+                                    color="primary"
+                                >
+                                    <FormattedMessage id="products.newProduct" defaultMessage="New Product" />
+                                </Button>
+                            }
+                            rowAction={(params) => (
+                                <IconButton component={StackLink} pageName="edit" payload={params.row.id}>
+                                    <Edit color="primary" />
+                                </IconButton>
+                            )}
```

Comment:
please create a copy of ProductsGrid for the two use cases to avoid having this customisation here at the page level